### PR TITLE
Add mutation tests

### PR DESCRIPTION
**Pull request by dwatl**
Output example
> Generated 1468 mutations Killed 516 (35%)
> Mutations with no coverage 768. Test strength 74%
> Ran 3296 tests (2.25 tests per mutation

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -63,6 +73,12 @@ tasks.test {
     }
 }
 
+tasks.withType<info.solidsoft.gradle.pitest.PitestTask> {
```
Comment (dagguh) [inline build.gradle.kts:76]: Let's make `check` task depend on this one

File: build.gradle.kts [hunk 2]
```diff
@@ -63,6 +73,12 @@ tasks.test {
...
+    threads.set(12)
+    outputFormats.set(setOf("XML", "HTML"))
```
Comment (dagguh) [inline build.gradle.kts:78]: Let's add these reports as GitHub Actions artifacts.

**Review by dagguh — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (dagguh) [reply build.gradle.kts]: Ok, `check` runs `pitest`:
```
» ./gradlew check --dry-run
Starting a Gradle Daemon (subsequent builds will be faster)
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:compileTestJava SKIPPED
:processTestResources SKIPPED
:testClasses SKIPPED
:test SKIPPED
:pitest SKIPPED
:testIntegration SKIPPED
:verifyLicensing SKIPPED
:check SKIPPED
```

**Review by dagguh — APPROVED** [review]

File: build.gradle.kts [hunk 3]
```diff
@@ -63,6 +73,12 @@ tasks.test {
     }
 }
 
+val pitest = tasks.withType<info.solidsoft.gradle.pitest.PitestTask> {
```
Comment (dagguh) [inline build.gradle.kts:76]: Tests failed:
```
10:24:00 AM PIT >> INFO : MINION : java.lang.AssertionError: [thruput per ms] 
10:24:00 AM PIT >> INFO : MINION : Expecting:
10:24:00 AM PIT >> INFO : MINION :  <1978.9049f>
10:24:00 AM PIT >> INFO : MINION : to be greater than:
10:24:00 AM PIT >> INFO : MINION :  <2500.0f> 
```
It seems the mutations are slowing down the benchmark. I guess we can drop the threshold?

Comment (dagguh) [reply build.gradle.kts:76]: This plugin is spamming the stdout. A `./gradlew build` should be basically empty unless there's an actionable for the dev, like a red test or a warning.

- [ ] reduce/remove stdout spam

**Review by dagguh — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 3 above)
Comment (dwatl) [reply build.gradle.kts:76]: We could exclude test from mutation. It's not checking the logic, but performance. Mutations are more useful for checking uncovered logical cases.

**Pull request closed without merging**